### PR TITLE
[SEVER-269] adding documentation for managing cluster user access in eks

### PR DESCRIPTION
**Pull request by soulchips**
This document outline how a cluster admin may manage providing cluster access to other team members

### CODE REVIEW

File: eks/managing_user_access.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+}
+```
+
+Upon `terrafrom apply`, these fields will whitelist your IPs, and add the list of users will be added to the system:masters group for full access to all resources in the cluster.
+
+### Updating the Cluster admin list
+If you wish to add/remove admin users from your existing cluster, you only need to update the values in your `terraform.tfvars` and run `terrafrom apply`.
+
+## Adding Users with Limited Resource Access
+You may not wish for each user to have complete access of to all cluster resources. To more finely tune user access we will make use kubernetes role based access control ([RBAC]).
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:31]: Either "we will make use of" or just "we will use"

File: eks/managing_user_access.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+In the following example we will create a role which will have limited access to a `develop` namespace and then map it to an IAM user. You will need to be a cluster administrator as detailed above to proceed with the following steps.
+
+We'll first we'll need to add the user's ARN to the aws-auth configmap in the kube-system namespace.
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:34]: Garbled beginning of sentence. Also stay consistent with (not) using contractions – I personally lean towards not using them in written language

**Review by robinsturmeit — CHANGES_REQUESTED** [review]
Two comments

File: eks/managing_user_access.md (hunk 1 above)
Comment (soulchips) [reply eks/managing_user_access.md:31]: ah good catch. thanks

File: eks/managing_user_access.md (hunk 2 above)
Comment (soulchips) [reply eks/managing_user_access.md:34]: another good catch :)

File: eks/managing_user_access.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+  },
+  {
+      groups   = ["system:masters"]
+      userarn  = "<user1 ARN>"
+      username = "<user1name>"
+  }
+}
+```
+
+Upon `terrafrom apply`, these fields will whitelist your IPs, and add the list of users will be added to the system:masters group for full access to all resources in the cluster.
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:25]: This sentence doesn't seem right

File: eks/managing_user_access.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+### Updating the Cluster admin list
+If you wish to add/remove admin users from your existing cluster, you only need to update the values in your `terraform.tfvars` and run `terrafrom apply`.
+
+## Adding Users with Limited Resource Access
+You may not wish for each user to have complete access of to all cluster resources. To more finely tune user access we make use kubernetes' role based access control ([RBAC]).
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:31]: Of vs to, "to tune ... more finely", "make use of"

File: eks/managing_user_access.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+- apiGroups: ["extensions","apps"]
+  resources: ["deployments"]
+  verbs: ["get", "list", "watch"]
+```
+
+- and then apply the role to your cluster:
+`kubectl apply -f read-role.yaml`
+
+
+3. Now we'll create a `role-binding` on our user's ARN and the pod-reader role in the cluster to define their access level.
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:77]: There's still a contraction

File: eks/managing_user_access.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+- add the following to a file call `read-role-binding.yaml`
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:78]: 'called'

File: eks/managing_user_access.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+roleRef:
+  kind: Role
+  name: pod-reader
+  apiGroup: rbac.authorization.k8s.io
+```
+
+- and then apply the role-binding to your cluster:
+`kubectl apply -f read-role-binding.yaml`
+
+4. Finally to grant access to the cluster, you will need to whitelist the user's IP address.
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:98]: Missing comma

File: eks/managing_user_access.md [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+- As with admin users, we will need to add the user's IP to your terrafrom.tfvars and `terraform apply`
+
+Now your user will be able to access the cluster but will only be able to view pod resources in the develop namespace.
+
+For more details on managing permissions, you may read the google's [RBAC] documentation.
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:103]: "You may read Google's" or "you may read the"

File: eks/managing_user_access.md [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+
+
+## Providing Access with IAM Groups
+You may not wish to repeat the process above for each user you wish to add. Many user's will require the same access levels. We can limit the repitition by assigning users to IAM groups.
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:107]: "users" not "user's"

**Review by robinsturmeit — CHANGES_REQUESTED** [review]
Found some more.

File: eks/managing_user_access.md (hunk 3 above)
Comment (soulchips) [reply eks/managing_user_access.md:25]: the line probably isn't necessary to i'll remove

File: eks/managing_user_access.md [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+For more details on managing permissions, you may read Google's [RBAC] documentation.
+
+
+## Providing Access with IAM Groups
+You may not wish to repeat the process above for each user you wish to add. Many users will require the same access levels. We can limit the repitition by assigning users to IAM groups.
+Similary to users, groups may also be mapped to roles to provide access management.
+
+1. First create an IAM role:
+```
+POLICY=$(echo -n '{"Version":"2012-10-17","Statement":[{"Effect":"Allow","Principal":{"AWS":"arn:aws:iam::'; echo -n "$ACCOUNT_ID"; echo -n ':root"},"Action":"sts:AssumeRole","Condition":{}}]}')
```
Comment (robinsturmeit) [inline eks/managing_user_access.md:111]: Can we be sure that $ACCOUNT_ID is set here and has the correct value?

**Review by robinsturmeit — COMMENTED** [review]
Didn't find any language issues, but had a technical question

File: eks/managing_user_access.md (hunk 10 above)
Comment (soulchips) [reply eks/managing_user_access.md:112]: made this a bit more explicit

**Review by robinsturmeit — APPROVED** [review]
Satisfied now! Thank you, I know it can be frustrating if you have to get back to an open PR that often.

**Comment by soulchips** [comment]
lol not a problem @robinsturmeit public facing docs should be tight

**Comment by robinsturmeit** [comment]
> lol not a problem @robinsturmeit public facing docs should be tight

My thoughts exactly 👍

**Review by ryanwohara — APPROVED** [review]

**Pull request merged**